scheduler/README.md

```diff
@@ -0,0 +1,192 @@
+# Nomad Scheduler
+
+This package holds the logic behind Nomad schedulers. The `Scheduler` interface
+is implemented by two objects:
+
+- `GenericScheduler` and
+- `SystemScheduler`.
+
+The `CoreScheduler` object also implements this interface, but it's use is
+purely internal, the core scheduler does not schedule any user jobs.
+
+Nomad scheduler's task is to, given an evaluation, produce a plan of placing the
+desired allocations on feasibile nodes. Consult [Nomad documentation][0] for
+more details.
+
+The diagram below illustrates this process for the service and system schedulers
+in more detail:
+
+```
+                                 ┌──────────────┐        ┌───────────┐                                               ┌─────────────┐      ┌──────────┐
+                                 │   cluster    │        │feasibility│                          ┌─────────────┐      │    score    │      │   plan   │
+    Service and batch jobs:      │reconciliation│───────▶│   check   │─────────────────────────▶│     fit     │─────▶│ assignment  │─────▶│submission│
```

Comment:
It looks like this is slightly misaligned. (Same in the row below.)

Reply:
I think perhaps that's a font setting or GH setting of sorts? I see it perfectly aligned.

Reply:
Weird. In the [rendered version](https://github.com/hashicorp/nomad/blob/736034781d79d8c3d3dedeeb3583a7f6312cab4c/scheduler/README.md) I see this:

![Screenshot From 2025-06-04 14-35-44](https://github.com/user-attachments/assets/993ac801-5f9d-44f2-b7c4-49efa8e79bb3)

Reply:
[7b23729](https://github.com/hashicorp/nomad/pull/25986/commits/7b23729fff18144dbcaf4a699dd783a5758cff57)